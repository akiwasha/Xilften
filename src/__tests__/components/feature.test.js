import React from 'react';
import { render } from '@testing-library/react';
import { Feature } from '../../components';

describe('<Feature />', () => {
  it('renders the <Feature /> with populated data', () => {
    const { container, getByText } = render(
      <Feature>
        <Feature.Title>
          Films, séries TV et bien plus en illimité.
        </Feature.Title>
        <Feature.SubTitle>
          Où que vous soyez. Annulez à tout moment.
        </Feature.SubTitle>
      </Feature>
    );

    expect(
      getByText('Films, séries TV et bien plus en illimité.')
    ).toBeTruthy();
    expect(getByText('Où que vous soyez. Annulez à tout moment.')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Feature /> with a title', () => {
    const { container, queryByText } = render(
      <Feature>
        <Feature.Title>
          Films, séries TV et bien plus en illimité.
        </Feature.Title>
      </Feature>
    );

    expect(
      queryByText('Films, séries TV et bien plus en illimité.')
    ).toBeTruthy();
    expect(
      queryByText('Où que vous soyez. Annulez à tout moment.')
    ).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Feature /> with a subtitle', () => {
    const { container, queryByText, getByText } = render(
      <Feature>
        <Feature.SubTitle>
          Où que vous soyez. Annulez à tout moment.
        </Feature.SubTitle>
      </Feature>
    );

    expect(
      queryByText('Films, séries TV et bien plus en illimité.')
    ).toBeFalsy();
    expect(
      queryByText('Où que vous soyez. Annulez à tout moment.')
    ).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
