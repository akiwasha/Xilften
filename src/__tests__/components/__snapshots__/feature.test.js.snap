// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders the <Feature /> with a subtitle 1`] = `
<div
  class="feature__Container-sc-1u94x48-0 enxrVh"
>
  <h2
    class="feature__SubTitle-sc-1u94x48-2 ckNrBq"
  >
    Où que vous soyez. Annulez à tout moment.
  </h2>
</div>
`;

exports[`<Feature /> renders the <Feature /> with a title 1`] = `
<div
  class="feature__Container-sc-1u94x48-0 enxrVh"
>
  <h1
    class="feature__Title-sc-1u94x48-1 gnETZS"
  >
    Films, séries TV et bien plus en illimité.
  </h1>
</div>
`;

exports[`<Feature /> renders the <Feature /> with populated data 1`] = `
<div
  class="feature__Container-sc-1u94x48-0 enxrVh"
>
  <h1
    class="feature__Title-sc-1u94x48-1 gnETZS"
  >
    Films, séries TV et bien plus en illimité.
  </h1>
  <h2
    class="feature__SubTitle-sc-1u94x48-2 ckNrBq"
  >
    Où que vous soyez. Annulez à tout moment.
  </h2>
</div>
`;
